realm/realm-library/src/main/cpp/io_realm_SyncManager.cpp

```diff
@@ -58,24 +53,14 @@ struct AndroidClientListener : public realm::BindingCallbackThreadObserver {
     }
 } s_client_thread_listener;
 
-JNIEXPORT void JNICALL Java_io_realm_SyncManager_nativeInitializeSyncClient(JNIEnv* env, jclass)
-{
-    TR_ENTER()
-    if (sync_client) {
-        return;
-    }
-
-    try {
-        // Setup SyncManager
-        g_binding_callback_thread_observer = &s_client_thread_listener;
-
-        // Create SyncClient
-        sync::Client::Config config;
-        config.logger = &CoreLoggerBridge::shared();
-        sync_client = std::make_unique<Client>(std::move(config)); // Throws
+struct AndroidSyncLoggerFactory : public realm::SyncLoggerFactory {
+    virtual std::unique_ptr<util::Logger> make_logger(Logger::Level level) override
+    {
+        auto logger = std::make_unique<CoreLoggerBridge>();
+        logger->set_level_threshold(level);
+        return std::move(logger);
```

Comment:
no need to call `std::move` here, compiler will optimize the code to avoid constructors here. It will just return a rvalue. See http://stackoverflow.com/questions/17473753/c11-return-value-optimization-or-move